src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -1780,45 +1780,51 @@ public bool IsOnForeachHeader(SyntaxNode root, int position, out SyntaxNode fore
             return IsOnHeader(position, node, node.CloseParenToken);
         }
 
-        public bool IsBetweenTypeMembers(SourceText sourceText, SyntaxNode root, int position)
+        public bool IsBetweenTypeMembers(SourceText sourceText, SyntaxNode root, int position, out SyntaxNode typeDeclaration)
         {
             var token = root.FindToken(position);
             var typeDecl = token.GetAncestor<TypeDeclarationSyntax>();
-            if (typeDecl != null)
+            typeDeclaration = typeDecl;
+
+            if (typeDecl == null)
+            {
+                return false;
+            }
+
+            if (position < typeDecl.OpenBraceToken.Span.End ||
+                position > typeDecl.CloseBraceToken.Span.Start)
+            {
+                return false;
+            }
+
+            var line = sourceText.Lines.GetLineFromPosition(position);
+            if (!line.IsEmptyOrWhitespace())
+            {
+                return false;
+            }
+
+            var member = typeDecl.Members.FirstOrDefault(d => d.FullSpan.Contains(position));
+            if (member == null)
             {
-                if (position >= typeDecl.OpenBraceToken.Span.End &&
-                    position <= typeDecl.CloseBraceToken.Span.Start)
+                // There are no members, or we're after the last member.
+                return true;
+            }
+            else
+            {
+                // We're within a member.  Make sure we're in the leading whitespace of
+                // the member.
+                if (position < member.SpanStart)
                 {
-                    var line = sourceText.Lines.GetLineFromPosition(position);
-                    if (!line.IsEmptyOrWhitespace())
+                    foreach (var trivia in member.GetLeadingTrivia())
                     {
-                        return false;
-                    }
+                        if (!trivia.IsWhitespaceOrEndOfLine())
```

Comment:
i feel like this will make the following now work:

```c#
class C
{

#region whatever
  $$
  int Foo() => 0;

#endregion

}
```

You'll hit the directive trivia and state that you're not between members (when you are).

Reply:
All of changes here were made via "InvertIf" refactoring so it should behave _exactly_ the same way as before. 

Will investigate, however.

Reply:
note: it's quite possible the code before had this problem.  I think what we actually want to say is "is the position after the start of the member, but before the first comment-trivia if it has one".

Reply:
Yeah, this is likely something i missed.  But hey, coding wouldnt' be coding if you didn't have to fixup the mistakes of the person who did the original stuff :)

Reply:
I think it should be fine, tho. The `!IsWhiteSpaceOrEndOfLine()` will not trigger.

![image](https://user-images.githubusercontent.com/1764721/63470660-0b5bab80-c422-11e9-803d-f04528e00557.png)

Reply:
> The !IsWhiteSpaceOrEndOfLine() will not trigger if I understand it correctly.

I think it will though.... because a RegionDirective is not a "WhitespaceOrEndOfLine".  So it will return false here.  Right?